### PR TITLE
fix undefined fields for cesc

### PR DESCRIPTION
**Pull request by trungnguyenexp**

### CODE REVIEW

**Comment by wmoussa** [comment]
what is the trello card that tirggered this work?

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ var priorTimeStamp = '0';
 var priorMarketingCode = "";
 var length = marketingChannels.length;
 
+
+// Old logic that goes through cesc fields to find last touch and then set DTL.
```
Comment (wmoussa) [inline src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:9]: if old logic, why is it still getting executed?

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js [hunk 2]
```diff
@@ -17,34 +20,44 @@ for(var i=0; i < length; i++){
...
 if(fullMarketingCode != undefined && patrialMarketingCode != undefined && fullMarketingCode.indexOf(patrialMarketingCode) > -1 ){
     var marketingCodeAndDTL = fullMarketingCode.split(patrialMarketingCode + ".");
     if( marketingCodeAndDTL.length >= 2 ){
         b["cescDTL"]  = marketingCodeAndDTL[1];
-        b["cescLastTouchMarketingCode"] = patrialMarketingCode;     //This is to handle the case where there is DTL in the lastTouchMarketingCode.
     }
 }
+
+
+// Use CID for last touch
```
Comment (wmoussa) [inline src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:39]: don't use acronym, CID could be mean anything..

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js [hunk 3]
```diff
@@ -17,34 +20,44 @@ for(var i=0; i < length; i++){
...
+    b['cescLastTouch_isexpired'] = true;
+}
+
+// Load rule looks for country and network, and if these are undefined in the udo the load rule will break and tag will not fire even it other conditions are met.
+// Therefore, initializing it to empty so load rule doesn't break.
+b["cescChannel"] = "";
+b["cescCompany"] = "";
+b["cescCountry"] = "";
+b["cescNetwork"] = "";
+if(b["cescLastTouchMarketingCode"].split(".").length >= 4){
```
Comment (wmoussa) [inline src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:58]: how do you guarantee cescLastTouchMarketingCode is defined?

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js [hunk 4]
```diff
@@ -17,34 +20,44 @@ for(var i=0; i < length; i++){
...
+    b["cescChannel"] = b["cescLastTouchMarketingCode"].split(".")[0];
```
Comment (wmoussa) [inline src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:59]: you should just prepare the array once, instead of splitting each time

**Review by wmoussa — CHANGES_REQUESTED** [review]

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js (hunk 1 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:9]: This comment has always been there. This PR showed that I added it but I didn't. I guess things get shifted around when I copy the code to a text editor to modify and then copy it back to the extension.

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js (hunk 2 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:39]: Same as above. I didn't add it but will find out what it stands for and update it.

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js (hunk 3 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:58]: Same as above, I didn't add 58, only line 54-57.

File: src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js (hunk 4 above)
Comment (trungnguyenexp) [reply src/library/simplified/extension/cesc-LT-channel-country-network-DTL.js:59]: Same as above, I didn't add this, it has been like this for a while, but I can update.

**Comment by trungnguyenexp** [comment]
For this PR, I only added lines 30, 54-57 where I initialized these fields so they won't break the load rules that have conditions to call them. 

It appears that this PR causes a lot of confusion and hard to read due to copy and pasty. I will close this one and will close another one that is more readable.

**Comment by trungnguyenexp** [comment]
new PR here https://github.com/ExpediaInc/expedia-omg-tags/pull/86

**Comment by trungnguyenexp** [comment]
new PR here: https://github.com/ExpediaInc/expedia-omg-tags/pull/86

**Pull request closed without merging**